Go panic, crash on v2.0.1

### Do you want to request a *feature* or report a *bug*?

Bug

### Did you try using a 1.7.x configuration for the version 2.0?

- [ ] Yes
- [X] No

### What did you do?
I am running Jira behind traefik v2.0.1. When I go to https://jira.example.com I get a go panic and traefik crashes.

```
panic: interface conversion: *buffer.bufferWriter is not http.Flusher: missing method Flush

goroutine 789 [running]:
github.com/containous/traefik/v2/pkg/middlewares/metrics.(*responseRecorder).Flush(0xc0004e17e0)
        /go/src/github.com/containous/traefik/pkg/middlewares/metrics/recorder.go:36 +0x43
github.com/containous/traefik/v2/pkg/middlewares/pipelining.(*writerWithoutCloseNotify).Flush(0xc000405260)
        /go/src/github.com/containous/traefik/pkg/middlewares/pipelining/pipelining.go:64 +0x68
net/http/httputil.(*maxLatencyWriter).delayedFlush(0xc000809620)
        /usr/local/go/src/net/http/httputil/reverseproxy.go:481 +0x88
created by time.goFunc
        /usr/local/go/src/time/sleep.go:168 +0x44
```

### What did you expect to see?

No crash

### What did you see instead?

A crash

### Output of `traefik version`: (_What version of Traefik are you using?_)
```
Version:      2.0.1
Codename:     montdor
Go version:   go1.13.1
Built:        2019-09-26T16:18:03Z
OS/Arch:      linux/amd64
```
### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```toml
# traefik.toml

[global]
  checkNewVersion = true
  sendAnonymousUsage = false

[entryPoints]
  [entryPoints.web]
    address = ":80"
  [entryPoints.web-secure]
    address = ":443"
  [entryPoints.postgres]
    address = ":5432"

[certificatesResolvers.letsencrypt.acme]
  email = "admin@example.co"
  storage = "/etc/letsencrypt/acme.json"
  #caServer = "https://acme-staging-v02.api.letsencrypt.org/directory"
  [certificatesResolvers.letsencrypt.acme.dnsChallenge]
    provider = "cloudflare"
    delayBeforeCheck = 0
    resolvers = ["jim.ns.cloudflare.com:53", "lia.ns.cloudflare.com:53"]

[api]
  dashboard = true

[ping]

[metrics]
  [metrics.prometheus]

[providers]
  [providers.docker]
    endpoint = "unix:///var/run/docker.sock"
    defaultRule = "Host(`{{ normalize .Name }}.example.com`)"
    watch = true
    exposedByDefault = false
  [providers.file]
    watch = true
    directory = "/etc/traefik/config"
```

Here is my jira ansible role with traefik labels:

```yml
- name: Create jira container
  docker_container:
    name: jira
    image: cptactionhank/atlassian-jira
    restart_policy: unless-stopped
    pull: yes
    volumes:
      - /opt/apps/jira:/var/atlassian/jira
    labels:
      traefik.enable: "true"
      traefik.http.middlewares.jira_limit.buffering.maxRequestBodyBytes: "10000000"
      traefik.http.routers.jira_https.entrypoints: web-secure
      traefik.http.routers.jira_https.middlewares: jira_limit
      traefik.http.routers.jira_https.rule: "Host(`{{ jira_fqdn }}`)"
      traefik.http.routers.jira_https.tls: "true"
      traefik.http.routers.jira_https.tls.certResolver: letsencrypt
      traefik.http.routers.jira_https.tls.domains[0].main: "{{ jira_fqdn }}"
      traefik.http.routers.jira_https.service: jira
      traefik.http.services.jira.loadbalancer.server.port: "8080"
    env:
      X_PROXY_NAME: "{{ jira_fqdn }}"
      X_PROXY_SCHEME: https
```

### If applicable, please paste the log output in DEBUG level (`--log.level=DEBUG` switch)

I will see what I can do. I am getting the crash on every request to that endpoint. 

```
time="2019-09-28T03:16:57Z" level=debug msg="vulcand/oxy/buffer: begin ServeHttp on request" Request="{\"Method\":\"GET\",\"URL\":{\"Scheme\":\"\",\"Opaque\":\"\",\"User\":null,\"Host\":\"\",\"Path\":\"/secure/Dashboard.jspa\",\"RawPath\":\"\",\"ForceQuery\":false,\"RawQuery\":\"\",\"Fragment\":\"\"},\"Proto\":\"HTTP/2.0\",\"ProtoMajor\":2,\"ProtoMinor\":0,\"Header\":{\"Accept\":[\"text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3\"],\"Accept-Encoding\":[\"gzip, deflate, br\"],\"Accept-Language\":[\"en-US,en;q=0.9\"],\"Cache-Control\":[\"max-age=0\"],\"Cookie\":[\"seraph.rememberme.cookie=10900%3Af954f28545287b3f5166cdb4e7c7dc66b97d1a94; atlassian.xsrf.token=BUDE-VWTC-D4A5-W7TD_b702e2fa0dc614625bf1a5a1a2b5cd2869ad39f3_lin; JSESSIONID=8B1C4EFA8438D9828F38F31B53B200E5\"],\"Dnt\":[\"1\"],\"Sec-Fetch-Mode\":[\"navigate\"],\"Sec-Fetch-Site\":[\"cross-site\"],\"Sec-Fetch-User\":[\"?1\"],\"Upgrade-Insecure-Requests\":[\"1\"],\"User-Agent\":[\"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.75 Safari/537.36\"],\"X-Forwarded-Host\":[\"jira.example.com\"],\"X-Forwarded-Port\":[\"443\"],\"X-Forwarded-Proto\":[\"https\"],\"X-Forwarded-Server\":[\"cae62a445a8b\"],\"X-Real-Ip\":[\"10.5.1.4\"]},\"ContentLength\":0,\"TransferEncoding\":null,\"Host\":\"jira.example.com\",\"Form\":null,\"PostForm\":null,\"MultipartForm\":null,\"Trailer\":null,\"RemoteAddr\":\"10.5.1.4:8574\",\"RequestURI\":\"/secure/Dashboard.jspa\",\"TLS\":null}"
time="2019-09-28T03:16:57Z" level=debug msg="vulcand/oxy/roundrobin/rr: begin ServeHttp on request" Request="{\"Method\":\"GET\",\"URL\":{\"Scheme\":\"\",\"Opaque\":\"\",\"User\":null,\"Host\":\"\",\"Path\":\"/secure/Dashboard.jspa\",\"RawPath\":\"\",\"ForceQuery\":false,\"RawQuery\":\"\",\"Fragment\":\"\"},\"Proto\":\"HTTP/2.0\",\"ProtoMajor\":2,\"ProtoMinor\":0,\"Header\":{\"Accept\":[\"text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3\"],\"Accept-Encoding\":[\"gzip, deflate, br\"],\"Accept-Language\":[\"en-US,en;q=0.9\"],\"Cache-Control\":[\"max-age=0\"],\"Cookie\":[\"seraph.rememberme.cookie=10900%3Af954f28545287b3f5166cdb4e7c7dc66b97d1a94; atlassian.xsrf.token=BUDE-VWTC-D4A5-W7TD_b702e2fa0dc614625bf1a5a1a2b5cd2869ad39f3_lin; JSESSIONID=8B1C4EFA8438D9828F38F31B53B200E5\"],\"Dnt\":[\"1\"],\"Sec-Fetch-Mode\":[\"navigate\"],\"Sec-Fetch-Site\":[\"cross-site\"],\"Sec-Fetch-User\":[\"?1\"],\"Upgrade-Insecure-Requests\":[\"1\"],\"User-Agent\":[\"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.75 Safari/537.36\"],\"X-Forwarded-Host\":[\"jira.example.com\"],\"X-Forwarded-Port\":[\"443\"],\"X-Forwarded-Proto\":[\"https\"],\"X-Forwarded-Server\":[\"cae62a445a8b\"],\"X-Real-Ip\":[\"10.5.1.4\"]},\"ContentLength\":0,\"TransferEncoding\":null,\"Host\":\"jira.example.com\",\"Form\":null,\"PostForm\":null,\"MultipartForm\":null,\"Trailer\":null,\"RemoteAddr\":\"10.5.1.4:8574\",\"RequestURI\":\"/secure/Dashboard.jspa\",\"TLS\":null}"
time="2019-09-28T03:16:57Z" level=debug msg="vulcand/oxy/roundrobin/rr: Forwarding this request to URL" Request="{\"Method\":\"GET\",\"URL\":{\"Scheme\":\"\",\"Opaque\":\"\",\"User\":null,\"Host\":\"\",\"Path\":\"/secure/Dashboard.jspa\",\"RawPath\":\"\",\"ForceQuery\":false,\"RawQuery\":\"\",\"Fragment\":\"\"},\"Proto\":\"HTTP/2.0\",\"ProtoMajor\":2,\"ProtoMinor\":0,\"Header\":{\"Accept\":[\"text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3\"],\"Accept-Encoding\":[\"gzip, deflate, br\"],\"Accept-Language\":[\"en-US,en;q=0.9\"],\"Cache-Control\":[\"max-age=0\"],\"Cookie\":[\"seraph.rememberme.cookie=10900%3Af954f28545287b3f5166cdb4e7c7dc66b97d1a94; atlassian.xsrf.token=BUDE-VWTC-D4A5-W7TD_b702e2fa0dc614625bf1a5a1a2b5cd2869ad39f3_lin; JSESSIONID=8B1C4EFA8438D9828F38F31B53B200E5\"],\"Dnt\":[\"1\"],\"Sec-Fetch-Mode\":[\"navigate\"],\"Sec-Fetch-Site\":[\"cross-site\"],\"Sec-Fetch-User\":[\"?1\"],\"Upgrade-Insecure-Requests\":[\"1\"],\"User-Agent\":[\"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.75 Safari/537.36\"],\"X-Forwarded-Host\":[\"jira.example.com\"],\"X-Forwarded-Port\":[\"443\"],\"X-Forwarded-Proto\":[\"https\"],\"X-Forwarded-Server\":[\"cae62a445a8b\"],\"X-Real-Ip\":[\"10.5.1.4\"]},\"ContentLength\":0,\"TransferEncoding\":null,\"Host\":\"jira.example.com\",\"Form\":null,\"PostForm\":null,\"MultipartForm\":null,\"Trailer\":null,\"RemoteAddr\":\"10.5.1.4:8574\",\"RequestURI\":\"/secure/Dashboard.jspa\",\"TLS\":null}" ForwardURL="http://172.17.0.7:8080"
time="2019-09-28T03:16:57Z" level=debug msg="vulcand/oxy/roundrobin/rr: completed ServeHttp on request" Request="{\"Method\":\"GET\",\"URL\":{\"Scheme\":\"\",\"Opaque\":\"\",\"User\":null,\"Host\":\"\",\"Path\":\"/secure/Dashboard.jspa\",\"RawPath\":\"\",\"ForceQuery\":false,\"RawQuery\":\"\",\"Fragment\":\"\"},\"Proto\":\"HTTP/2.0\",\"ProtoMajor\":2,\"ProtoMinor\":0,\"Header\":{\"Accept\":[\"text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3\"],\"Accept-Encoding\":[\"gzip, deflate, br\"],\"Accept-Language\":[\"en-US,en;q=0.9\"],\"Cache-Control\":[\"max-age=0\"],\"Cookie\":[\"seraph.rememberme.cookie=10900%3Af954f28545287b3f5166cdb4e7c7dc66b97d1a94; atlassian.xsrf.token=BUDE-VWTC-D4A5-W7TD_b702e2fa0dc614625bf1a5a1a2b5cd2869ad39f3_lin; JSESSIONID=8B1C4EFA8438D9828F38F31B53B200E5\"],\"Dnt\":[\"1\"],\"Sec-Fetch-Mode\":[\"navigate\"],\"Sec-Fetch-Site\":[\"cross-site\"],\"Sec-Fetch-User\":[\"?1\"],\"Upgrade-Insecure-Requests\":[\"1\"],\"User-Agent\":[\"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.75 Safari/537.36\"],\"X-Forwarded-Host\":[\"jira.example.com\"],\"X-Forwarded-Port\":[\"443\"],\"X-Forwarded-Proto\":[\"https\"],\"X-Forwarded-Server\":[\"cae62a445a8b\"],\"X-Real-Ip\":[\"10.5.1.4\"]},\"ContentLength\":0,\"TransferEncoding\":null,\"Host\":\"jira.example.com\",\"Form\":null,\"PostForm\":null,\"MultipartForm\":null,\"Trailer\":null,\"RemoteAddr\":\"10.5.1.4:8574\",\"RequestURI\":\"/secure/Dashboard.jspa\",\"TLS\":null}"
time="2019-09-28T03:16:57Z" level=debug msg="vulcand/oxy/buffer: completed ServeHttp on request" Request="{\"Method\":\"GET\",\"URL\":{\"Scheme\":\"\",\"Opaque\":\"\",\"User\":null,\"Host\":\"\",\"Path\":\"/secure/Dashboard.jspa\",\"RawPath\":\"\",\"ForceQuery\":false,\"RawQuery\":\"\",\"Fragment\":\"\"},\"Proto\":\"HTTP/2.0\",\"ProtoMajor\":2,\"ProtoMinor\":0,\"Header\":{\"Accept\":[\"text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3\"],\"Accept-Encoding\":[\"gzip, deflate, br\"],\"Accept-Language\":[\"en-US,en;q=0.9\"],\"Cache-Control\":[\"max-age=0\"],\"Cookie\":[\"seraph.rememberme.cookie=10900%3Af954f28545287b3f5166cdb4e7c7dc66b97d1a94; atlassian.xsrf.token=BUDE-VWTC-D4A5-W7TD_b702e2fa0dc614625bf1a5a1a2b5cd2869ad39f3_lin; JSESSIONID=8B1C4EFA8438D9828F38F31B53B200E5\"],\"Dnt\":[\"1\"],\"Sec-Fetch-Mode\":[\"navigate\"],\"Sec-Fetch-Site\":[\"cross-site\"],\"Sec-Fetch-User\":[\"?1\"],\"Upgrade-Insecure-Requests\":[\"1\"],\"User-Agent\":[\"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.75 Safari/537.36\"],\"X-Forwarded-Host\":[\"jira.example.com\"],\"X-Forwarded-Port\":[\"443\"],\"X-Forwarded-Proto\":[\"https\"],\"X-Forwarded-Server\":[\"cae62a445a8b\"],\"X-Real-Ip\":[\"10.5.1.4\"]},\"ContentLength\":0,\"TransferEncoding\":null,\"Host\":\"jira.example.com\",\"Form\":null,\"PostForm\":null,\"MultipartForm\":null,\"Trailer\":null,\"RemoteAddr\":\"10.5.1.4:8574\",\"RequestURI\":\"/secure/Dashboard.jspa\",\"TLS\":null}"
time="2019-09-28T03:16:57Z" level=debug msg="vulcand/oxy/buffer: begin ServeHttp on request" Request="{\"Method\":\"GET\",\"URL\":{\"Scheme\":\"\",\"Opaque\":\"\",\"User\":null,\"Host\":\"\",\"Path\":\"/s/5a8dd217c5ab2161145be5d1323373a6-CDN/ur7kyp/804002/2e3527675227261b74c45af3b1391de1/1.0/_/download/batch/jira.webresources:global-static-adgs/jira.webresources:global-static-adgs.css\",\"RawPath\":\"\",\"ForceQuery\":false,\"RawQuery\":\"\",\"Fragment\":\"\"},\"Proto\":\"HTTP/2.0\",\"ProtoMajor\":2,\"ProtoMinor\":0,\"Header\":{\"Accept\":[\"text/css,*/*;q=0.1\"],\"Accept-Encoding\":[\"gzip, deflate, br\"],\"Accept-Language\":[\"en-US,en;q=0.9\"],\"Cookie\":[\"seraph.rememberme.cookie=10900%3Af954f28545287b3f5166cdb4e7c7dc66b97d1a94; atlassian.xsrf.token=BUDE-VWTC-D4A5-W7TD_b702e2fa0dc614625bf1a5a1a2b5cd2869ad39f3_lin; JSESSIONID=8B1C4EFA8438D9828F38F31B53B200E5\"],\"Dnt\":[\"1\"],\"Referer\":[\"https://jira.example.com/secure/Dashboard.jspa\"],\"Sec-Fetch-Mode\":[\"no-cors\"],\"Sec-Fetch-Site\":[\"same-origin\"],\"User-Agent\":[\"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.75 Safari/537.36\"],\"X-Forwarded-Host\":[\"jira.example.com\"],\"X-Forwarded-Port\":[\"443\"],\"X-Forwarded-Proto\":[\"https\"],\"X-Forwarded-Server\":[\"cae62a445a8b\"],\"X-Real-Ip\":[\"10.5.1.4\"]},\"ContentLength\":0,\"TransferEncoding\":null,\"Host\":\"jira.example.com\",\"Form\":null,\"PostForm\":null,\"MultipartForm\":null,\"Trailer\":null,\"RemoteAddr\":\"10.5.1.4:8574\",\"RequestURI\":\"/s/5a8dd217c5ab2161145be5d1323373a6-CDN/ur7kyp/804002/2e3527675227261b74c45af3b1391de1/1.0/_/download/batch/jira.webresources:global-static-adgs/jira.webresources:global-static-adgs.css\",\"TLS\":null}"
time="2019-09-28T03:16:57Z" level=debug msg="vulcand/oxy/roundrobin/rr: begin ServeHttp on request" Request="{\"Method\":\"GET\",\"URL\":{\"Scheme\":\"\",\"Opaque\":\"\",\"User\":null,\"Host\":\"\",\"Path\":\"/s/5a8dd217c5ab2161145be5d1323373a6-CDN/ur7kyp/804002/2e3527675227261b74c45af3b1391de1/1.0/_/download/batch/jira.webresources:global-static-adgs/jira.webresources:global-static-adgs.css\",\"RawPath\":\"\",\"ForceQuery\":false,\"RawQuery\":\"\",\"Fragment\":\"\"},\"Proto\":\"HTTP/2.0\",\"ProtoMajor\":2,\"ProtoMinor\":0,\"Header\":{\"Accept\":[\"text/css,*/*;q=0.1\"],\"Accept-Encoding\":[\"gzip, deflate, br\"],\"Accept-Language\":[\"en-US,en;q=0.9\"],\"Cookie\":[\"seraph.rememberme.cookie=10900%3Af954f28545287b3f5166cdb4e7c7dc66b97d1a94; atlassian.xsrf.token=BUDE-VWTC-D4A5-W7TD_b702e2fa0dc614625bf1a5a1a2b5cd2869ad39f3_lin; JSESSIONID=8B1C4EFA8438D9828F38F31B53B200E5\"],\"Dnt\":[\"1\"],\"Referer\":[\"https://jira.example.com/secure/Dashboard.jspa\"],\"Sec-Fetch-Mode\":[\"no-cors\"],\"Sec-Fetch-Site\":[\"same-origin\"],\"User-Agent\":[\"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.75 Safari/537.36\"],\"X-Forwarded-Host\":[\"jira.example.com\"],\"X-Forwarded-Port\":[\"443\"],\"X-Forwarded-Proto\":[\"https\"],\"X-Forwarded-Server\":[\"cae62a445a8b\"],\"X-Real-Ip\":[\"10.5.1.4\"]},\"ContentLength\":0,\"TransferEncoding\":null,\"Host\":\"jira.example.com\",\"Form\":null,\"PostForm\":null,\"MultipartForm\":null,\"Trailer\":null,\"RemoteAddr\":\"10.5.1.4:8574\",\"RequestURI\":\"/s/5a8dd217c5ab2161145be5d1323373a6-CDN/ur7kyp/804002/2e3527675227261b74c45af3b1391de1/1.0/_/download/batch/jira.webresources:global-static-adgs/jira.webresources:global-static-adgs.css\",\"TLS\":null}"
time="2019-09-28T03:16:57Z" level=debug msg="vulcand/oxy/roundrobin/rr: Forwarding this request to URL" Request="{\"Method\":\"GET\",\"URL\":{\"Scheme\":\"\",\"Opaque\":\"\",\"User\":null,\"Host\":\"\",\"Path\":\"/s/5a8dd217c5ab2161145be5d1323373a6-CDN/ur7kyp/804002/2e3527675227261b74c45af3b1391de1/1.0/_/download/batch/jira.webresources:global-static-adgs/jira.webresources:global-static-adgs.css\",\"RawPath\":\"\",\"ForceQuery\":false,\"RawQuery\":\"\",\"Fragment\":\"\"},\"Proto\":\"HTTP/2.0\",\"ProtoMajor\":2,\"ProtoMinor\":0,\"Header\":{\"Accept\":[\"text/css,*/*;q=0.1\"],\"Accept-Encoding\":[\"gzip, deflate, br\"],\"Accept-Language\":[\"en-US,en;q=0.9\"],\"Cookie\":[\"seraph.rememberme.cookie=10900%3Af954f28545287b3f5166cdb4e7c7dc66b97d1a94; atlassian.xsrf.token=BUDE-VWTC-D4A5-W7TD_b702e2fa0dc614625bf1a5a1a2b5cd2869ad39f3_lin; JSESSIONID=8B1C4EFA8438D9828F38F31B53B200E5\"],\"Dnt\":[\"1\"],\"Referer\":[\"https://jira.example.com/secure/Dashboard.jspa\"],\"Sec-Fetch-Mode\":[\"no-cors\"],\"Sec-Fetch-Site\":[\"same-origin\"],\"User-Agent\":[\"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.75 Safari/537.36\"],\"X-Forwarded-Host\":[\"jira.example.com\"],\"X-Forwarded-Port\":[\"443\"],\"X-Forwarded-Proto\":[\"https\"],\"X-Forwarded-Server\":[\"cae62a445a8b\"],\"X-Real-Ip\":[\"10.5.1.4\"]},\"ContentLength\":0,\"TransferEncoding\":null,\"Host\":\"jira.example.com\",\"Form\":null,\"PostForm\":null,\"MultipartForm\":null,\"Trailer\":null,\"RemoteAddr\":\"10.5.1.4:8574\",\"RequestURI\":\"/s/5a8dd217c5ab2161145be5d1323373a6-CDN/ur7kyp/804002/2e3527675227261b74c45af3b1391de1/1.0/_/download/batch/jira.webresources:global-static-adgs/jira.webresources:global-static-adgs.css\",\"TLS\":null}" ForwardURL="http://172.17.0.7:8080"
time="2019-09-28T03:16:57Z" level=debug msg="vulcand/oxy/buffer: begin ServeHttp on request" Request="{\"Method\":\"GET\",\"URL\":{\"Scheme\":\"\",\"Opaque\":\"\",\"User\":null,\"Host\":\"\",\"Path\":\"/s/71321f22423b354566506405f4b2a8b4-CDN/ur7kyp/804002/2e3527675227261b74c45af3b1391de1/1.0/_/download/batch/jira.webresources:global-static/jira.webresources:global-static.css\",\"RawPath\":\"\",\"ForceQuery\":false,\"RawQuery\":\"\",\"Fragment\":\"\"},\"Proto\":\"HTTP/2.0\",\"ProtoMajor\":2,\"ProtoMinor\":0,\"Header\":{\"Accept\":[\"text/css,*/*;q=0.1\"],\"Accept-Encoding\":[\"gzip, deflate, br\"],\"Accept-Language\":[\"en-US,en;q=0.9\"],\"Cookie\":[\"seraph.rememberme.cookie=10900%3Af954f28545287b3f5166cdb4e7c7dc66b97d1a94; atlassian.xsrf.token=BUDE-VWTC-D4A5-W7TD_b702e2fa0dc614625bf1a5a1a2b5cd2869ad39f3_lin; JSESSIONID=8B1C4EFA8438D9828F38F31B53B200E5\"],\"Dnt\":[\"1\"],\"Referer\":[\"https://jira.example.com/secure/Dashboard.jspa\"],\"Sec-Fetch-Mode\":[\"no-cors\"],\"Sec-Fetch-Site\":[\"same-origin\"],\"User-Agent\":[\"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.75 Safari/537.36\"],\"X-Forwarded-Host\":[\"jira.example.com\"],\"X-Forwarded-Port\":[\"443\"],\"X-Forwarded-Proto\":[\"https\"],\"X-Forwarded-Server\":[\"cae62a445a8b\"],\"X-Real-Ip\":[\"10.5.1.4\"]},\"ContentLength\":0,\"TransferEncoding\":null,\"Host\":\"jira.example.com\",\"Form\":null,\"PostForm\":null,\"MultipartForm\":null,\"Trailer\":null,\"RemoteAddr\":\"10.5.1.4:8574\",\"RequestURI\":\"/s/71321f22423b354566506405f4b2a8b4-CDN/ur7kyp/804002/2e3527675227261b74c45af3b1391de1/1.0/_/download/batch/jira.webresources:global-static/jira.webresources:global-static.css\",\"TLS\":null}"
time="2019-09-28T03:16:57Z" level=debug msg="vulcand/oxy/roundrobin/rr: begin ServeHttp on request" Request="{\"Method\":\"GET\",\"URL\":{\"Scheme\":\"\",\"Opaque\":\"\",\"User\":null,\"Host\":\"\",\"Path\":\"/s/71321f22423b354566506405f4b2a8b4-CDN/ur7kyp/804002/2e3527675227261b74c45af3b1391de1/1.0/_/download/batch/jira.webresources:global-static/jira.webresources:global-static.css\",\"RawPath\":\"\",\"ForceQuery\":false,\"RawQuery\":\"\",\"Fragment\":\"\"},\"Proto\":\"HTTP/2.0\",\"ProtoMajor\":2,\"ProtoMinor\":0,\"Header\":{\"Accept\":[\"text/css,*/*;q=0.1\"],\"Accept-Encoding\":[\"gzip, deflate, br\"],\"Accept-Language\":[\"en-US,en;q=0.9\"],\"Cookie\":[\"seraph.rememberme.cookie=10900%3Af954f28545287b3f5166cdb4e7c7dc66b97d1a94; atlassian.xsrf.token=BUDE-VWTC-D4A5-W7TD_b702e2fa0dc614625bf1a5a1a2b5cd2869ad39f3_lin; JSESSIONID=8B1C4EFA8438D9828F38F31B53B200E5\"],\"Dnt\":[\"1\"],\"Referer\":[\"https://jira.example.com/secure/Dashboard.jspa\"],\"Sec-Fetch-Mode\":[\"no-cors\"],\"Sec-Fetch-Site\":[\"same-origin\"],\"User-Agent\":[\"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.75 Safari/537.36\"],\"X-Forwarded-Host\":[\"jira.example.com\"],\"X-Forwarded-Port\":[\"443\"],\"X-Forwarded-Proto\":[\"https\"],\"X-Forwarded-Server\":[\"cae62a445a8b\"],\"X-Real-Ip\":[\"10.5.1.4\"]},\"ContentLength\":0,\"TransferEncoding\":null,\"Host\":\"jira.example.com\",\"Form\":null,\"PostForm\":null,\"MultipartForm\":null,\"Trailer\":null,\"RemoteAddr\":\"10.5.1.4:8574\",\"RequestURI\":\"/s/71321f22423b354566506405f4b2a8b4-CDN/ur7kyp/804002/2e3527675227261b74c45af3b1391de1/1.0/_/download/batch/jira.webresources:global-static/jira.webresources:global-static.css\",\"TLS\":null}"
time="2019-09-28T03:16:57Z" level=debug msg="vulcand/oxy/roundrobin/rr: Forwarding this request to URL" Request="{\"Method\":\"GET\",\"URL\":{\"Scheme\":\"\",\"Opaque\":\"\",\"User\":null,\"Host\":\"\",\"Path\":\"/s/71321f22423b354566506405f4b2a8b4-CDN/ur7kyp/804002/2e3527675227261b74c45af3b1391de1/1.0/_/download/batch/jira.webresources:global-static/jira.webresources:global-static.css\",\"RawPath\":\"\",\"ForceQuery\":false,\"RawQuery\":\"\",\"Fragment\":\"\"},\"Proto\":\"HTTP/2.0\",\"ProtoMajor\":2,\"ProtoMinor\":0,\"Header\":{\"Accept\":[\"text/css,*/*;q=0.1\"],\"Accept-Encoding\":[\"gzip, deflate, br\"],\"Accept-Language\":[\"en-US,en;q=0.9\"],\"Cookie\":[\"seraph.rememberme.cookie=10900%3Af954f28545287b3f5166cdb4e7c7dc66b97d1a94; atlassian.xsrf.token=BUDE-VWTC-D4A5-W7TD_b702e2fa0dc614625bf1a5a1a2b5cd2869ad39f3_lin; JSESSIONID=8B1C4EFA8438D9828F38F31B53B200E5\"],\"Dnt\":[\"1\"],\"Referer\":[\"https://jira.example.com/secure/Dashboard.jspa\"],\"Sec-Fetch-Mode\":[\"no-cors\"],\"Sec-Fetch-Site\":[\"same-origin\"],\"User-Agent\":[\"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.75 Safari/537.36\"],\"X-Forwarded-Host\":[\"jira.example.com\"],\"X-Forwarded-Port\":[\"443\"],\"X-Forwarded-Proto\":[\"https\"],\"X-Forwarded-Server\":[\"cae62a445a8b\"],\"X-Real-Ip\":[\"10.5.1.4\"]},\"ContentLength\":0,\"TransferEncoding\":null,\"Host\":\"jira.example.com\",\"Form\":null,\"PostForm\":null,\"MultipartForm\":null,\"Trailer\":null,\"RemoteAddr\":\"10.5.1.4:8574\",\"RequestURI\":\"/s/71321f22423b354566506405f4b2a8b4-CDN/ur7kyp/804002/2e3527675227261b74c45af3b1391de1/1.0/_/download/batch/jira.webresources:global-static/jira.webresources:global-static.css\",\"TLS\":null}" ForwardURL="http://172.17.0.7:8080"
panic: interface conversion: *buffer.bufferWriter is not http.Flusher: missing method Flush

goroutine 115 [running]:
github.com/containous/traefik/v2/pkg/middlewares/metrics.(*responseRecorder).Flush(0xc00000ce60)
        /go/src/github.com/containous/traefik/pkg/middlewares/metrics/recorder.go:36 +0x43
github.com/containous/traefik/v2/pkg/middlewares/pipelining.(*writerWithoutCloseNotify).Flush(0xc0004d0090)
        /go/src/github.com/containous/traefik/pkg/middlewares/pipelining/pipelining.go:64 +0x68
net/http/httputil.(*maxLatencyWriter).delayedFlush(0xc0003caa50)
        /usr/local/go/src/net/http/httputil/reverseproxy.go:481 +0x88
created by time.goFunc
        /usr/local/go/src/time/sleep.go:168 +0x44
```

--- a/pkg/middlewares/metrics/recorder.go
+++ b/pkg/middlewares/metrics/recorder.go
@@ -33,5 +33,7 @@ func (r *responseRecorder) CloseNotify() <-chan bool {
 
 // Flush sends any buffered data to the client.
 func (r *responseRecorder) Flush() {
-	r.ResponseWriter.(http.Flusher).Flush()
+	if f, ok := r.ResponseWriter.(http.Flusher); ok {
+		f.Flush()
+	}
 }

